### PR TITLE
Document new spectral cube-related functionality in specutils

### PR DESCRIPTION
**Pull request by ibusko**

### CODE REVIEW

**Comment by PatrickOgle** [comment]
I don't see the spectral_slab doc in Files changed for this PR, so I can't complete the requested review.

**Comment by ibusko** [comment]
@PatrickOgle I can see it 
image

File: docs/analysis.rst [hunk 1]
```diff
@@ -134,6 +134,23 @@ While this example is "pre-subtracted", this function only performs well if the
 contiuum has already been subtracted, as for the other functions above and
 below.
 
+
+Moment
+--------
```
Comment (nmearl) [inline docs/analysis.rst:139]:
```suggestion
------
```

File: docs/spectral_regions.rst [hunk 2]
```diff
@@ -209,6 +209,44 @@ An example of a multiple sub-region `~specutils.SpectralRegion`:
     >>> sub_spectra[1].spectral_axis
     <SpectralAxis [34., 35., 36., 37., 38., 39., 40.] nm>
 
+
+The bounding region that includes all data, including the ones that lie
+in between disjoint spectral regions, can be extracted with
```
Comment (nmearl) [inline docs/spectral_regions.rst:214]:
```suggestion
in between disjointed spectral regions, can be extracted with
```

**Review by nmearl — CHANGES_REQUESTED** [review]

**Review by PatrickOgle — APPROVED** [review]
Documentation on new functionality looks good.

**Comment by rosteen** [comment]
The doc build failure is:

```
Warning, treated as error:
/home/runner/work/specutils/specutils/docs/spectral_regions.rst:213:py:obj reference target not found: extract_bounding_spectral_region
```

Maybe it needs the full package reference `specutils.manipulation.extract_bounding_spectral_region` there instead of just the function name?

**Comment by ibusko** [comment]
This is superseded by #783

**Pull request closed without merging**